theano/gpuarray/tests/check_dnn.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+## This command should make this script stop at first error occured
+## or keyboard interruption received.
+set -e
+
+## Get all DNN algorithms supported by Theano.
+
+export SUPPORTED_DNN_CONV_ALGO_FWD=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_ALGO_RUNTIME, SUPPORTED_DNN_CONV_ALGO_FWD; print('\n'.join(SUPPORTED_DNN_CONV_ALGO_FWD + SUPPORTED_DNN_CONV_ALGO_RUNTIME))")
+
+export SUPPORTED_DNN_CONV_ALGO_BWD_DATA=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_ALGO_RUNTIME, SUPPORTED_DNN_CONV_ALGO_BWD_DATA; print('\n'.join(SUPPORTED_DNN_CONV_ALGO_BWD_DATA + SUPPORTED_DNN_CONV_ALGO_RUNTIME))")
+
+export SUPPORTED_DNN_CONV_ALGO_BWD_FILTER=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_ALGO_RUNTIME, SUPPORTED_DNN_CONV_ALGO_BWD_FILTER; print('\n'.join(SUPPORTED_DNN_CONV_ALGO_BWD_FILTER + SUPPORTED_DNN_CONV_ALGO_RUNTIME))")
+
+export SUPPORTED_DNN_CONV_PRECISION=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_PRECISION; print('\n'.join(SUPPORTED_DNN_CONV_PRECISION))")
+
+## List special Theano DNN tests that depend on DNN algorithms.
+## Other Theano DNN tests will be run once.
+## These special Theano DNN tests will be run 
+## for every combination of DNN algorithms.
+
+declare -a special_tests=('theano.gpuarray.tests.test_dnn.test_conv3d_bwd'
+                          'theano.gpuarray.tests.test_dnn.test_conv3d_fwd'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_alpha_output_merge'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_border_mode'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_grad'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_inplace'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_merge'
+                          'theano.gpuarray.tests.test_dnn.TestDnnInferShapes')
```

Comment:
There are ones in theano/gpuarray/tests/test_abstractconv.py as well (TestDnnConv2d, TestDnnConv3d).